lib/ansible/modules/network/cloudengine/ce_startup.py

```diff
@@ -0,0 +1,468 @@
+
+        # software_file check
+        if self.software_file:
+            if not self.get_software_filename_type(self.software_file):
+                self.module.fail_json(
+                    msg='Error: Invalid software file name or software file name extension ( *.cc).')
+
+        # patch_file check
+        if self.patch_file:
+            if not self.get_pat_filename_type(self.patch_file):
+                self.module.fail_json(
+                    msg='Error: Invalid patch file name or patch file name extension ( *.PAT ).')
+
+        # slot check
+        if self.slot:
+            if self.slot.isdigit():
+                if int(self.slot) <= 0 or int(self.slot) > 16:
+                    self.module.fail_json(
+                        msg='Error: The number of slot is not in the range from 1 to 16.')
+            else:
+                if len(self.slot) <= 0 or len(self.slot) > 32:
+                    self.module.fail_json(
+                        msg='Error: The length of slot is not in the range from 1 to 32.')
+
+    def get_proposed(self):
+        """get proposed info"""
+
+        if self.cfg_file:
+            self.proposed["cfg_file"] = self.cfg_file
+        if self.software_file:
+            self.proposed["system_file"] = self.software_file
+        if self.patch_file:
+            self.proposed["patch_file"] = self.patch_file
+        if self.slot:
+            self.proposed["slot"] = self.slot
+
+    def get_existing(self):
+        """get existing info"""
+
+        if not self.startup_info:
+            return
+        self.existing["StartupInfos"] = self.startup_info["StartupInfos"]
+
+    def get_end_state(self):
+        """get end state info"""
+        self.end_state["StartupInfos"] = None
+
+    def work(self):
+        """worker"""
+
+        self.check_params()
+        self.get_proposed()
+        self.startup_info = self.get_startup_dict()
+        self.get_existing()
+        if self.cfg_file:
+            self.startup_next_cfg_file()
+        if self.software_file:
+            self.startup_next_software_file()
+        if self.patch_file:
+            self.startup_next_pat_file()
+        if self.action == "display":
+            self.startup_info = self.get_startup_dict()
+
+        self.get_end_state()
+        self.results['changed'] = self.changed
+        self.results['proposed'] = self.proposed
+        self.results['existing'] = self.existing
+        self.results['end_state'] = self.end_state
+        if self.changed:
+            self.results['updates'] = self.updates_cmd
+        else:
+            self.results['updates'] = list()
+
+        self.module.exit_json(**self.results)
+
+
+def main():
+    """ Module main """
+
+    argument_spec = dict(
+        cfg_file=dict(type='str'),
+        software_file=dict(type='str'),
+        patch_file=dict(type='str'),
+        slot=dict(type='str'),
```

Comment:
For integers please:
use `type='int`
remove the `isdigit` check from `check_params`